R/data_mile2.R

```diff
@@ -0,0 +1,38 @@
+#' FIMS input data frame for milestone 2
+#'
+#' A dataset containing information necessary to run an age-structured stock
+#' assessment model with length composition data in FIMS for milestone 2 . 
+#' This data was generated using
+#' the `ASSAMC` package written for the [model comparison project](
+#' www.github.com/Bai-Li-NOAA/Age_Structured_Stock_Assessment_Model_Comparison).
+#'
+#' @format A data frame with `r NROW(data_mile2)` observations of
+#' `r NCOL(data_mile2)` variables:
+#' \describe{
+#'   \item{type}{The type of data the row contains. Allowed types include
+#'     `age`, `length-comp`, `length-at-age`, `index`, `landings`, and `weight-at-age` data.}
```

Comment:
Why name it "length-comp" when age-composition data are just "age"? Also, can the model fit to "length-at-age" data at this point in time? If no, then we should remove it from the list.

Reply:
I think this is probably because it was age-comp in the original dev version I was working off. What was the reasoning for changing to age instead of age-comp? I feel like this is confusing given that age is already a column of the data frame and I imagine there may be other types of data we want to add at some point that are also age related?

Reply:
The model does not currently fit to length-at-age they are parameters though I put them in here so that are available for the vignette similar to the empirical weight-at-age values. I think eventually it would be useful to have these as inputs that parameters can be estimated from though to allow some error.

Reply:
Regarding "length-at-age" ... disregard my comment, I was thinking conditional length-at-age data. I think that aybe we just need to explain better what each type is to be more clear in general so people like me don't make wrong assumptions 😉

Reply:
it looks like NROW and NCOL are capitalized because that is how those functions are named. If I hover over them vscode says base::NCOL

Reply:
I cleaned these up to be age-comp and age-to-length-conversion. I also changed to name of weight-at-age to be population rather than fleet1 because it applies to everything. This will also be clearer if/when we add in fleet specific weight at age capabilities